events/api.py

```diff
@@ -1483,18 +1482,58 @@ def get_has_regular_users(obj):
         return obj.regular_users.count() > 0
 
 
+class OrganizationUsersField(serializers.SlugRelatedField):
+    def __init__(self, **kwargs):
+        super().__init__(queryset=User.objects.all(), slug_field="username", **kwargs)
+
+    def to_representation(self, obj):
+        return UserSerializer(obj).data
+
+
 class OrganizationDetailSerializer(OrganizationListSerializer):
-    regular_users = serializers.SerializerMethodField()
-    admin_users = serializers.SerializerMethodField()
+    user_fields = ["admin_users", "regular_users"]
 
-    def validate(self, data):
-        if Organization.objects.filter(id=str(self.request.data["id"])):
-            raise DRFPermissionDenied(
-                f"Organization with id {self.request.data['id']} already exists."
-            )
-        return super().validate(data)
+    admin_users = OrganizationUsersField(
+        many=True,
+        required=False,
+    )
+
+    regular_users = OrganizationUsersField(
+        many=True,
+        required=False,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.instance:
+            self.fields["data_source"].read_only = True
+            self.fields["origin_id"].read_only = True
+
+    def to_representation(self, obj):
+        ret = super().to_representation(obj)
+
+        user = self.context["user"]
+        # Show admin users and regular users only to admins
+        user_fields = set(self.user_fields)
+        if user.is_anonymous or not user.is_admin_of(obj):
+            for field in user_fields:
+                del ret[field]
```

Comment:
Imho you can remove this and put the following to `__init__`.
```python
        user = self.context["user"]
        # Show admin users and regular users only to admins
        if user.is_anonymous or not user.is_admin_of(obj):
            for field in self.user_fields:
                self.fields.pop(field, None)
```

Reply:
Fixed. Added also unit tests to make sure that regular user and anonymous user cannot see admin_users and regular_users